d2bs/kolbot/libs/common/Loader.js

```diff
@@ -105,16 +107,17 @@ var Loader = {
 								reconfiguration = typeof Scripts[i] === 'object';
 
 								if (typeof (global[i]) === "function") {
-									if (i !== "Test" && i !== "Follower") {
-										townCheck = Town.goToTown();
-									} else {
+									if (this.skipTown.indexOf(i) > -1) {
 										townCheck = true;
+									} else {
+										townCheck = Town.goToTown();
 									}
 
 									if (townCheck) {
 										print("ÿc2Starting script: ÿc9" + i);
 										//scriptBroadcast(JSON.stringify({currScript: i}));
 										Messaging.sendToScript("tools/toolsthread.js", JSON.stringify({currScript: i}));
+										this.scriptIndex++;
```

Comment:
consider the following refactor:
```
		for (s in Scripts) {
			if (Scripts[s]) {
				scriptList.push(s);
			}
		}

		for (this.scriptIndex = 0; this.scriptIndex < charScripts.length; this.scriptIndex++) {
			var script = this.scriptList[this.scriptIndex];

			if (this.fileList.indexOf(script) < 0) {
				Misc.errorReport("ÿc1Script " + script + " doesn't exist.");
				continue;
			}

			if (!include("bots/" + script + ".js")) {
				Misc.errorReport("Failed to include script: " + script);
				continue;
			}

			if (isIncluded("bots/" + script + ".js")) {
				try {
					reconfiguration = typeof Scripts[script] === 'object';

					if (typeof (global[script]) !== "function") {
						throw new Error("Invalid script function name");
					}

					if (this.skipTown.indexOf(script) > -1 || Town.goToTown()) {
						print("ÿc2Starting script: ÿc9" + script);
						//scriptBroadcast(JSON.stringify({currScript: script}));
						Messaging.sendToScript("tools/toolsthread.js", JSON.stringify({currScript: script}));

						if (reconfiguration) {
							print("ÿc2Copying Config properties from " + script + " object.");
							this.copy(Scripts[script], Config);
						}

						global[script]();

						if (reconfiguration) {
							print("ÿc2Reverting back unmodified config properties.");
							this.copy(unmodifiedConfig, Config);
						}
					}
				} catch (error) {
					Misc.errorReport(error, script);
				}
			}
		}
```

change existing scriptList to fileList
move charScripts from scriptName to main function, rename charScripts to scriptList
change the loader loop from what it is to for loop using new scriptList you have created

Reply:
looks good, modifications done